### ISSUE TITLE
CombinedResourceHandler make Primeface 6.0 can not write new resource for partial request

### ISSUE THREAD

**Issue by thojava** [omnifaces/omnifaces#273]
Hi BalusC, 

After i upgraded our system to PF 6.0 today, your CombinedResourceHandler seem not work well with the new version. Symptom is resources are required by ajax request is not encoded.

As i investigated, PrimePartialResponseWriter changed its algorithm on encoded dynamic resource, they only reload resource ( update <head> tag ) if number of current resources is greater than number of resources in last request. 

// dynamic resource loading
                ArrayList<ResourceUtils.ResourceInfo> initialResources = DynamicResourcesPhaseListener.getInitialResources(context);
                ArrayList<ResourceUtils.ResourceInfo> currentResources = ResourceUtils.getComponentResources(context);
                if (initialResources != null && currentResources != null && currentResources.size() > initialResources.size()) {}

The initialResources is evaluated at restored view before your handler executed.
While the current resourced have been combined by your handler so above condition likely failed.

Could you help to look on this issue, let me know if you have any enquiries

**Comment by BalusC** [comment]
Same problem is reported in BootsFaces: https://github.com/TheCoder4eu/BootsFaces-OSP/issues/411

**Comment by BalusC** [comment]
I tried creating a reproducer so I could debug it, but I can't reproduce the problem. I imagined conditionally rendering or even building new PrimeFaces components would trigger it, but it doesn't. What's the minimum required code to reproduce this problem?

**Comment by thojava** [comment]
In my case, I have problem with p:fileUpload component.
The scenario is, at first request system just display list of employee in table layout. p:fileUpload is not required to rendered.
User click to choose one employee, system navigate he to the employee detail page. In this page, we need to render p:fileUpload but the required resource used by p:fileUpload are fileUpload.css and fileUpload.js are not loaded due to the problem explained originally.

Let me know if you need more info

**Comment by BalusC** [comment]
So, navigating by ajax should trigger it? Sorry, I still can't reproduce it.

index.xhtml
```
<h:form>
    <p:commandButton value="navigate to upload" action="upload" />
</h:form>	
```
upload.xhtml
```
<h:form>
    <p:fileUpload />
</h:form>	
```
It just returns a partial response with `update id="javax.faces.ViewRoot"` with the correct script resources already enclosed.

**Comment by thojava** [comment]
Actually the summary and detail page i mentioned above are the same f:view. I conditionally render the summary or detail page. 

So in your case index.xhtml and upload.xhtml share the same f:view or not ?

**Comment by BalusC** [comment]
No, as you said it's a navigation.

**Comment by thojava** [comment]
Sorry, I has been used the term navigation wrongly.
So the problem is with ajax request within the same f:view and conditionally render p:fileUpload

**Comment by BalusC** [comment]
I can't do much without a reproducer.

**Comment by thojava** [comment]
Okay. I will give you the reproducer project tomorrow.

**Comment by BalusC** [comment]
Great, looking forward it.

**Comment by thojava** [comment]
[CombinedResourceTest.zip](https://github.com/omnifaces/omnifaces/files/336262/CombinedResourceTest.zip)

I have uploaded source project here, it is a maven one, running on Wildfly but hope that you can run it with tomcat as well.

When building this sample project I have found out more clue. 
The problem appear when I have added component and component resource dynamically at the PreRenderEventListener. You can easily check my source and comment there.

**Comment by mydeadlyvenoms** [comment]
Hi BalusC, since you referenced this issue [(https://github.com/TheCoder4eu/BootsFaces-OSP/issues/411)](https://github.com/TheCoder4eu/BootsFaces-OSP/issues/411) you may want to use [https://github.com/mydeadlyvenoms/BootsFacesIssues](https://github.com/mydeadlyvenoms/BootsFacesIssues) to reproduce the issue, it is maven based..

Just upgrade to PrimeFaces 6.0 and add the CombinedResourceHandler feature.

**Comment by BalusC** [comment]
I reproduced the problem and I have locally fixed it (just let `CombinedResourceHandler` skip its job when the current request is an ajax request with partial rendering). I will do some more intensive testing before committing the fix into 2.4.

**Comment by BalusC** [comment]
No bad things as to state management happened in both Mojarra 2.2.13 and MyFaces 2.2.10. Fix is available in today's latest [2.4-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/org/omnifaces/omnifaces/2.4-SNAPSHOT/). Please let me know if that works out for you.
